packages/mco/components/dr-status-popover/parsers/subscription-parser.tsx

```diff
@@ -0,0 +1,228 @@
+  const targetCluster = findCluster(drClusters, primaryClusterName);
+
+  const lastGroupSyncTime: string = drpc?.status?.lastGroupSyncTime;
+  const replicationType = getReplicationType(schedulingInterval);
+
+  const healthStatus = getReplicationHealth(
+    lastGroupSyncTime,
+    schedulingInterval,
+    replicationType
+  );
+
+  return {
+    drPolicyName,
+    schedulingInterval,
+    primaryCluster: primaryClusterName,
+    targetCluster: targetCluster?.metadata?.name,
+    lastGroupSyncTime,
+    healthStatus,
+    phase: drpc?.status?.phase,
+  };
+};
+
+const getDRResources = (
+  namespace: string
+): {
+  resources: {
+    drPolicies: ReturnType<typeof getDRPolicyResourceObj>;
+    drClusters: ReturnType<typeof getDRClusterResourceObj>;
+    drPlacementControls: ReturnType<typeof getDRPlacementControlResourceObj>;
+  };
+} => ({
+  resources: {
+    drPolicies: getDRPolicyResourceObj(),
+    drClusters: getDRClusterResourceObj(),
+    drPlacementControls: getDRPlacementControlResourceObj({
+      namespace,
+    }),
+  },
+});
+
+const getSubscriptionResources = (
+  application: ApplicationKind,
+  namespace: string,
+  drResources: DisasterRecoveryResourceKind,
+  drLoaded: boolean,
+  drLoadError: any
+): {
+  resources: {
+    subscriptions: ReturnType<typeof getSubscriptionResourceObj>;
+    placementRules: ReturnType<typeof getPlacementRuleResourceObj>;
+    placements: ReturnType<typeof getPlacementResourceObj>;
+    placementDecisions: ReturnType<typeof getPlacementDecisionsResourceObj>;
+  };
+  drResources: {
+    data: DisasterRecoveryResourceKind;
+    loaded: boolean;
+    loadError: any;
+  };
+  overrides: {
+    applications: {
+      data: ApplicationKind;
+      loaded: boolean;
+      loadError: string;
+    };
+  };
+} => ({
+  resources: {
+    subscriptions: getSubscriptionResourceObj({
+      namespace,
+    }),
+    placementRules: getPlacementRuleResourceObj({
+      namespace,
+    }),
+    placements: getPlacementResourceObj({
+      namespace,
+    }),
+    placementDecisions: getPlacementDecisionsResourceObj({
+      namespace,
+    }),
+  },
+  drResources: {
+    data: drResources,
+    loaded: drLoaded,
+    loadError: drLoadError,
+  },
+  overrides: {
+    applications: {
+      data: application,
+      loaded: true,
+      loadError: '',
+    },
+  },
+});
+
+export const SubscriptionParser: React.FC<SubscriptionParserProps> = ({
+  application,
+}) => {
+  const namespace = getNamespace(application);
+
+  const [drResources, drLoaded, drLoadError] = useDisasterRecoveryResourceWatch(
+    getDRResources(namespace)
+  );
+
+  const [subscriptionResourceList, subsResourceLoaded, subsResourceLoadError] =
+    useSubscriptionResourceWatch(
+      getSubscriptionResources(
+        application,
+        namespace,
+        drResources,
+        drLoaded,
+        drLoadError
+      )
+    );
+
+  const allLoaded: boolean =
+    drLoaded && subsResourceLoaded && !drLoadError && !subsResourceLoadError;
```

Comment:
```suggestion
  const isLoadedWOError: boolean =
    subsResourceLoaded && !subsResourceLoadError;
```
subsResourceLoaded and subsResourceLoadError already includes drLoaded and drLoadError